src/type.js

```diff
@@ -0,0 +1,199 @@
+'use strict';
+
+const typeValidators = {
+	any: () => true,
+	array: emitWarning(createPrimitiveTypeValidator('array')),
```

Comment:
I see what you're doing here in a functional approach, but both me and Eduardo would prefer to have a more verbose approach that would be easier to read. To understand how these functions are even created you have to go into these more generic factory functions.

Something more like this would be clearer:

```javascript
const typeValidators = {
  any: () => true,
  array: function(value, name, context) {
    return validateType(value, context, 'array');
  },
  bool: function(value, name, context) {
    return validateType(value, context, 'boolean');
  },
  ...
};

function validateType(value, context, expectedType) {
  const type = getStateType(value);
	if (type !== expectedType) {
		emitWarning(`Expected type ${expectedType}`, context);
	}
}

function emitWarning(error, context) {
	const componentName = context.constructor.name;
	const parentComponent = context.getRenderer ? context.getRenderer().lastParentComponent_ : null;
	const parentComponentName = parentComponent ? parentComponent.constructor.name : null;

	console.error(
		`Warning: Invalid state '${stateName}' of type '${getStateType(stateValue)}' ` +
		`passed to '${componentName}'. ${expectedError}.  Check render method of ${parentComponentName}.`
	);

	return true
}
```